### DISCUSSION TITLE
Can we reopen project folder as last open‘s, not user's home folder?

### DISCUSSION THREAD

**Discussion by ouliuquan** [eclipse-theia/theia#12511] [Ideas] [answered]
![image](https://github.com/eclipse-theia/theia/assets/38776890/6a098540-b25a-4ea3-af44-6cc6585ec2bc)

I think this is a useful tips.

**Comment by vince-fugnitto** [comment, marked as answer]
@ouliuquan do you mean when performing "open folder" you want to see the dialog opened at the last folder opened by the user? 

If so anything is possible but you likely want to customise the implementation of the "Open Folder" command in a custom extension:

https://github.com/eclipse-theia/theia/blob/d959af083ab98d8f7e982a69bac331fc48e18eca/packages/workspace/src/browser/workspace-frontend-contribution.ts#L139-L142

https://github.com/eclipse-theia/theia/blob/d959af083ab98d8f7e982a69bac331fc48e18eca/packages/workspace/src/browser/workspace-frontend-contribution.ts#L312-L331

We follow the default behavior of displaying the workspace root if a workspace is currently opened, and falling back to the user's home similarly to VS Code. In you custom extension you can try falling back to the previous workspace root if available.

**Reply by ouliuquan** [reply]
Sorry, I don't fully remember the relationship between this, I remember confusing it. Thank you so much for your patience in explaining.
